### PR TITLE
Scan PR - Add option to scan source and most common ancestor target c…

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
- [x] Update [documentation](https://github.com/jfrog/documentation) about new features / new supported technologies
---

### Scan PR/MR command
We are scanning both the source and target branches.
Then we compare the results and filter only those found in the source, not in the target.

The new option calculates the most common ancestor commit by running `git merge-base source target`, checking out to the commit, and treating it as the target to be merged to.
To use this option you can:
* Specify it as an env var: `JF_USE_MOST_COMMON_ANCESTOR_AS_TARGET`
* Specify it in Frogbot config: `useMostCommonAncestorAsTarget` (at `Git` section)

### CODE REVIEW

**Comment by eranturgeman** [comment]
fix title + consider adding an explanation about when one should use this feature

File: utils/git.go [hunk 1]
```diff
@@ -154,6 +155,64 @@ func (gm *GitManager) Checkout(branchName string) error {
 	return nil
 }
 
+func (gm *GitManager) CheckoutToHash(hash, targetBranchWd string) error {
+	log.Debug("Running git fetch...")
```
Comment (eranturgeman) [inline utils/git.go:159]: duplicate debug log, as similar one exists in Fetch itself

File: utils/git.go [hunk 2]
```diff
@@ -154,6 +155,64 @@ func (gm *GitManager) Checkout(branchName string) error {
...
+}
+
+func (gm *GitManager) Fetch() error {
+	log.Debug("Running git fetch...")
+	err := gm.localGitRepository.Fetch(&git.FetchOptions{
+		RemoteName: gm.remoteName,
+		RemoteURL:  gm.remoteGitUrl,
+		Auth:       gm.auth,
+	})
+	if err != nil && err != git.NoErrAlreadyUpToDate {
```
Comment (eranturgeman) [inline utils/git.go:177]: what happens when the error is NoErrAlreadyUpToDate? will the fetch is still performed when we got this error?

File: utils/git.go [hunk 3]
```diff
@@ -154,6 +155,64 @@ func (gm *GitManager) Checkout(branchName string) error {
...
+	// Get the commit of the base branch
+	baseCommitHash, err := gm.localGitRepository.ResolveRevision(plumbing.Revision(baseBranch))
+	if err != nil {
+		return "", err
+	}
+	baseCommit, err := gm.localGitRepository.CommitObject(*baseCommitHash)
+	if err != nil {
+		return "", err
+	}
+	// Get the commit of the head branch
```
Comment (eranturgeman) [inline utils/git.go:193]:
```suggestion
	// Get the HEAD commit of the target branch
```

File: utils/git.go [hunk 4]
```diff
@@ -201,6 +260,26 @@ func (gm *GitManager) CreateBranchAndCheckout(branchName string, keepLocalChange
 	return err
 }
 
+func (gm *GitManager) createBranchAndCheckoutToHash(hash string, keepLocalChanges bool) error {
+	var checkoutConfig *git.CheckoutOptions
+	if keepLocalChanges {
```
Comment (eranturgeman) [inline utils/git.go:265]: In the single usage of this new function you pass keepLocalChanges=false, so why do we need to even refer to that in this function?
Is this intended for future usage or plans? If not we can consider deleting it

**Comment by eranturgeman** [comment]
Where does useMostCommonAncestorAsTarget field be inserted in frogbot-config.yml? is it part of the 'Project' list or a general definition.
We need to consider if we even want to add it as a field to frogbot-config.yml since the config file is going to change soon and maybe we should avoid adding new field to it

In addition, please open a ticket to add this new field to ConfigProfile

**Review by eranturgeman — COMMENTED** [review]
Looks good, see my comments

File: utils/git.go (hunk 2 above)
Comment (attiasas) [reply utils/git.go:183]: the fetch fails since there is nothing to fetch, in this case, we want to continue, any other error will be returned.

**Pull request merged**